File: djJava/HelloJava06/src/javastudy03_abstract/Dog.java
package javastudy03_abstract;

import javastudy02_abstract.Animal;

public class Dog extends Animal {

	@Override
	public void sleep() {
		System.out.println("주인 곁에서 잠을 잡니다...");

	}

	@Override
	public void eat() {
		System.out.println("고개를 그릇에 두고 밥을 먹습니다.");

	}
	//주인 이름
	private String masterName;
	public String getMasterName() {
		return masterName;
	}
	public void setMasterName(String masterName) {
		this.masterName = masterName;
	}
	public void bark() {
		System.out.println("멍멍멍");
	}

}

File: djCSharp2/HelloCSharp004/HelloCSharp004/Dog.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HelloCSharp004
{
    public class Dog : Animal
    {
        //virtual : 자손 클래스가 해당 메서드를 오버라이드 하겠다는 의미
        //오버라이드 한다는 의미 : 덮어쓰겠단 의미
        public virtual void Bark()
        {
            Console.WriteLine("멍멍 짖습니다.");
        }

        public override void eat()
        {
            Console.WriteLine("개 답게 먹습니다.");
        }

        public override void fight()
        {
            Console.WriteLine("뭅니다.");
        }
    }
}
